specification/datamigration/resource-manager/Microsoft.DataMigration/preview/2022-11-30-preview/definitions/MigrateMySqlAzureDbForMySqlOfflineTask.json

```diff
@@ -0,0 +1,500 @@
+          "readOnly": true
+        },
+        "resultPrefix": {
+          "type": "string",
+          "description": "Wildcard string prefix to use for querying all sub-tem results of the item",
+          "readOnly": true
+        },
+        "lastStorageUpdate": {
+          "type": "string",
+          "format": "date-time",
+          "description": "Last time the storage was updated",
+          "readOnly": true
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "#/definitions/MigrateMySqlAzureDbForMySqlOfflineTaskOutput"
+        }
+      ]
+    },
+    "MigrateMySqlAzureDbForMySqlOfflineTaskOutputError": {
+      "type": "object",
+      "x-ms-discriminator-value": "ErrorOutput",
+      "properties": {
+        "id": {
+          "type": "string",
+          "description": "Unique identifier for the exception",
+          "readOnly": true
+        },
+        "error": {
+          "$ref": "./Common.json#/definitions/ReportableException",
+          "description": "Migration error",
+          "readOnly": true
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "#/definitions/MigrateMySqlAzureDbForMySqlOfflineTaskOutput"
+        }
+      ]
+    },
+    "MigrateMySqlAzureDbForMySqlOfflineDatabaseInput": {
+      "type": "object",
+      "description": "Database specific information for offline MySQL to Azure Database for MySQL migration task inputs",
+      "properties": {
+        "name": {
+          "type": "string",
+          "description": "Name of the database"
+        },
+        "targetDatabaseName": {
+          "type": "string",
+          "description": "Name of target database. Note: Target database will be truncated before starting migration."
+        },
+        "tableMap": {
+          "type": "object",
+          "description": "Mapping of source to target tables",
+          "additionalProperties": {
+            "type": "string"
+          }
+        },
+        "tablesToMigrateSchema": {
+          "type": "object",
+          "description": "Mapping of source to target tables for schema migration",
+          "additionalProperties": {
+            "type": "string"
+          }
+        },
+        "selectedViews": {
+          "type": "array",
+          "description": "List of views to migrate",
+          "items": {
+            "type": "string"
+          }
+        },
+        "selectedTriggers": {
+          "type": "array",
+          "description": "List of triggers to migrate",
+          "items": {
+            "type": "string"
+          }
+        },
+        "selectedRoutines": {
+          "type": "array",
+          "description": "List of routines to migrate",
+          "items": {
+            "type": "string"
+          }
+        },
+        "selectedEvents": {
+          "type": "array",
+          "description": "List of events to migrate",
+          "items": {
+            "type": "string"
+          }
```

Comment:
Do we need a required section here?

Reply:
Technically, the fields like the name and the target database name should be required, and they have always been, but adding required fields is a breaking change that needs a review with the Azure’s Breaking Change Review Board, according to the [policy](http://aka.ms/bcforapi). Since these are existing fields, I would rather not change it.